tests/testsuite/shell_completions/mod.rs

```diff
@@ -1,4 +1,22 @@
-#![allow(unused)]
+use std::path::PathBuf;
+
+use cargo::util::command_prelude::*;
+use cargo_test_support::cargo_test;
+
+#[cargo_test]
+fn test_get_registry_candidates() {
+    let current_dir = std::env::current_dir().expect("Failed to get current directory");
+    let cwd = PathBuf::from(file!()).parent().unwrap().join("template");
+    let cwd = current_dir.join(cwd);
```

Comment:
A temp directory is created for each test.  I'd recommend going ahead and writing to that and using that as the "cwd" for the test case.  We have `ProjectBuilder` to help in creating these.  You can also publish packages to a dev registry.